### PR TITLE
feat: Source detail version endpoint

### PR DESCRIPTION
**Pull request by hosekpeter**
Jira: [PSGO-779](https://keboola.atlassian.net/browse/PSGO-779)

**Changes:**
- added a new endpoint
- added tests

-----------

[PSGO-779]: https://keboola.atlassian.net/browse/PSGO-779?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/stream/design.go [hunk 1]
```diff
@@ -344,6 +344,19 @@ var _ = Service("stream", func() {
...
+	Method("SourceVersionDetail", func() {
+		Meta("openapi:summary", "Source version detail")
+		Description("Source version detail.")
+		Result(EntityVersion)
+		Payload(SourceVersionDetailRequest)
+		HTTP(func() {
+			GET("/branches/{branchId}/sources/{sourceId}/versions/{version_number}")
+			Meta("openapi:tag:configuration")
+			Response(StatusOK)
+			SourceNotFoundError()
```
Comment (jachym-tousek-keboola) [inline api/stream/design.go:356]: I'd expect `VersionNotFoundError()` as well.

File: api/stream/design.go [hunk 2]
```diff
@@ -344,6 +344,19 @@ var _ = Service("stream", func() {
 		})
 	})
 
+	Method("SourceVersionDetail", func() {
+		Meta("openapi:summary", "Source version detail")
+		Description("Source version detail.")
+		Result(EntityVersion)
+		Payload(SourceVersionDetailRequest)
+		HTTP(func() {
+			GET("/branches/{branchId}/sources/{sourceId}/versions/{version_number}")
```
Comment (jachym-tousek-keboola) [inline api/stream/design.go:353]: `{versionNumber}` for consistency.

File: api/stream/design.go (hunk 1 above)
Comment (hosekpeter) [reply api/stream/design.go:356]: [01bde4f](https://github.com/keboola/keboola-as-code/pull/2087/commits/01bde4f2e4982529ebc73bf07555954d24a9cdfa)

File: api/stream/design.go (hunk 2 above)
Comment (hosekpeter) [reply api/stream/design.go]: [01bde4f](https://github.com/keboola/keboola-as-code/pull/2087/commits/01bde4f2e4982529ebc73bf07555954d24a9cdfa)

**Review by Matovidlo — APPROVED** [review]
LGTM

File: test/stream/api/source/version-detail-404/003-patch-settings/request.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "path": "/v1/branches/default/sources/my-source/settings",
```
Comment (jachym-tousek-keboola) [inline test/stream/api/source/version-detail-404/003-patch-settings/request.json:2]: `version-detail-404` can be simplified. You don't need to patch settings, you aren't using the new version in the test anyway.

File: test/stream/api/source/version-detail-ok/004-patch-settings/request.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "path": "/v1/branches/default/sources/my-source/settings",
```
Comment (jachym-tousek-keboola) [inline test/stream/api/source/version-detail-ok/004-patch-settings/request.json:2]: Why is patch settings called twice in the e2e tests? (`003-patch-settings` and `004-patch-settings`)

**Review by jachym-tousek-keboola — CHANGES_REQUESTED** [review]

File: test/stream/api/source/version-detail-ok/004-patch-settings/request.json (hunk 4 above)
Comment (hosekpeter) [reply test/stream/api/source/version-detail-ok/004-patch-settings/request.json]: [0377fac](https://github.com/keboola/keboola-as-code/pull/2087/commits/0377facfe1520062da1e095c3d68e991ce59c69e)

File: test/stream/api/source/version-detail-404/003-patch-settings/request.json (hunk 3 above)
Comment (hosekpeter) [reply test/stream/api/source/version-detail-404/003-patch-settings/request.json]: [0377fac](https://github.com/keboola/keboola-as-code/pull/2087/commits/0377facfe1520062da1e095c3d68e991ce59c69e)

File: test/stream/api/source/version-detail-404/003-version-detail/expected-response.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "statusCode": 404,
+  "error": "stream.api.sourceVersionNotFound",
+  "message": "Source version \"my-source/0000000025\" not found in the branch."
```
Comment (jachym-tousek-keboola) [inline test/stream/api/source/version-detail-404/003-version-detail/expected-response.json:4]: There should be one more step in this test - try accessing version detail before creating the source. The message should be different in that case.

File: test/stream/api/source/version-detail-404/003-version-detail/expected-response.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+}
```
Comment (jachym-tousek-keboola) [inline test/stream/api/source/version-detail-404/003-version-detail/expected-response.json:5]: Missing new lines at the end of some files...

Comment (hosekpeter) [reply test/stream/api/source/version-detail-404/003-version-detail/expected-response.json]: 47f6964

File: test/stream/api/source/version-detail-404/003-version-detail/expected-response.json (hunk 5 above)
Comment (hosekpeter) [reply test/stream/api/source/version-detail-404/003-version-detail/expected-response.json]: 47f6964

**Review by jachym-tousek-keboola — APPROVED** [review]

**Pull request merged**